airbyte-workers/src/main/java/io/airbyte/workers/DefaultSyncWorker.java

```diff
@@ -49,17 +49,24 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(DefaultSyncWorker.class);
 
+  private final long jobId;
+  private final int attempt;
   private final Source<AirbyteMessage> source;
   private final Destination<AirbyteMessage> destination;
   private final MessageTracker<AirbyteMessage> messageTracker;
   private final NormalizationRunner normalizationRunner;
 
   private final AtomicBoolean cancelled;
 
-  public DefaultSyncWorker(final Source<AirbyteMessage> source,
+  public DefaultSyncWorker(
+                           final long jobId,
+                           final int attempt,
+                           final Source<AirbyteMessage> source,
                            final Destination<AirbyteMessage> destination,
                            final MessageTracker<AirbyteMessage> messageTracker,
                            final NormalizationRunner normalizationRunner) {
+    this.jobId = jobId;
+    this.attempt = attempt;
```

Comment:
Across the PR the tuple jobId/attempt should probably be passed as a data structure instead of two parameters. that will ensure that they never get separated and it will avoid parameters creep

Why not a class call WorkerContext? This way if we need to pass additional metadata about the run we can without having to touch all the codebase?